### ISSUE TITLE
Insufficient training set sample time, unclear task boundaries

### ISSUE THREAD

**Issue by elspru** [general-ai-challenge/Round1#7]
Hi, 
so I'm gathering sample data for the micro tasks, 
and it seems that running through the whole alphabet leads to only 2 reply-answer solutions before switching to a new task.
However to do any kind of training need to have at least 3 reply-answer solutions, 2 for training and 1 for testing. 

Here are some proposed solutions:
1. at least double the soft limit of the micro tasks
2. at least halve the alphabet size
3. (should be done anyways) make a clear indication that there is a task switch

3 should ideally be done in any case, would be easy if for instance all the capitals were taken out of the alphabet, and they were used exclusively for indicating the start of a class of task. (assuming there are less than or equal to 26 variations).  alternatively can simply have for instance A for a different soft-limit micro-task, and B for a different kind of task altogether.

**Comment by martinpoliak-goodai** [comment]
Hi elspru, 
you will notice that in later tasks (starting with task group 5), this answer-collecting approach stops working because randomly hitting the correct answer gets more and more difficult (for instance, you need to output a space followed by a correct number in task 5.1). 

We're helping the agent by starting with an easy tasks first (1-4) - in a single task instance, there always at least exists a one-to-one mapping between the character on input and the correct answer at that position. You could exploit this fact when you're collecting data for your initial training. But still, your agent will need to be able to cope with the more advanced tasks too, where hitting reward by acting randomly is nearly impossible.

Regarding option 3 - knowing that there was a task [instance] switch - we really prefer if the agent is able to recognize a task [instance] switch on its own, without explicitly being told so. Such an agent should be much more flexible when placed in an unfamiliar environment.

**Comment by elspru** [comment]
Yes Martin, you bring up a good point,

There is far too much dithering and changing of criteria,
it is really quite ridiculous how there is no standard that is simply expanded upon.
Instead the rules are changed many times without warning.
It's like having to learn a new language every 100 words. 

One input, one reward, and one feedback,seems to be the very best format. 
I think you should just stick to that for the whole thing. 
That would make it much easier to gather training data. 
Then wouldn't have to worry about the other things I mentioned. 

can't teach something if you don't have anything to train it with,
the random prodding is necessary in the begining now, precisely because there is no solution demonstrated.

**Comment by martinpoliak-goodai** [comment]
It's necessary to show a solver a diverse set of task instances for it to be able to generalize to a solution that works on all of the task instances. This means we actually need to teach a "new language every 100 words" so that the solver can find one unifying, abstract language by some form of generalization.

I think you would not perceive the instance switch as such an interruption if the instances were e.g. images and not temporal sequences. The problem with temporal sequences in real life is that if you have a sequence of them, there may be no clear indication where one temporal sequence ends and another begins. We made the design decision to keep this uncertainty in the tasks. So far we've had reports of multiple teams successfully solving (at least) tasks 1-4, which indicates that it's possible to overcome this uncertainty.

**Comment by elspru** [comment]
I agree with a diverse set of tasks, but it makes sense for each subtask to be viable for training.

It's easily possible (even though there may still be some wasted rounds) if you have knowledge regarding the length of rounds (or rows of pixels if you prefer) before a task switch,
while you give this information in the readme, I don't know that it will be the same  in the challenge environment.

**Comment by axaraspb** [comment]
We would like to remind you that we encourage the participants to extend the training datasets, and to use any additional training data you may find useful. This is not forbidden by the rules, quite the contrary: your ideas and suggestions are encouraged, and the outcomes will be considered for not only quantitative, but also qualitative evaluation.

The evaluation still will be performed on the original, non-disclosed evaluation curriculum created by General AI Challenge team.

Best,
Daria,
General AI Challenge team

**Comment by dariagoodai** [comment]
We would like to remind you that we encourage the participants to extend the training datasets, and to use any additional training data you may find useful. This is not forbidden by the rules, quite the contrary: your ideas and suggestions are encouraged, and the outcomes will be considered for not only quantitative, but also qualitative evaluation.

The evaluation still will be performed on the original, non-disclosed evaluation curriculum created by General AI Challenge team.

Best,
Daria,
General AI Challenge team
